packages/pages/src/components/address/methods.ts

```diff
@@ -22,3 +27,152 @@ export const getUnabbreviated = (
 
   return unabbreviatedField && address[unabbreviatedField];
 };
+
+/**
+ * Get a third-party maps url for a Yext location
+ *
+ * @param {any} profile - Partial of Yext Location entity profile
+ * @param {Provider} provider - Google, Apple, Bing
+ * @param {boolean} directions - Enable driving directions
+ *
+ * @returns {string} - Maps service url
+ */
+export const getDirections = (
+  profile: any,
+  provider?: MapProvider,
+  directions?: boolean
+): string | undefined => {
+  // TODO: replace profile type any with partial type declaration for profile
+  if (!profile.ref_listings && !profile.address) return undefined;
+  const { address } = profile;
+
+  let query = encodeArray([
+    address?.line1,
+    address?.line2,
+    address?.city,
+    address?.region,
+    address?.postalCode,
+    address?.countryCode,
+  ]);
+
+  switch (provider) {
+    case "APPLE":
+      return getDirectionsApple(query, directions);
+      break;
+    case "BING":
+      query = encodeArray([
+        address?.line1,
+        address?.city,
+        address?.region,
+        address?.postalCode,
+      ]);
+      return getDirectionsBing(query, directions);
+      break;
+    default:
+      const gmbURL = getlistingUrl(
```

Comment:
It's little weird you construct the gmbURL and then check if it exists. Can you determine if you should construct/return it beforehand? Like check if ref_listings then return gmb, else if googlePlaceId return getDirectionsGooglePlaceID(), else return getDirectionsGoogle()

Reply:
I changed the method to just get the `ListingType` so it is more obvious that it has to see if a listings is available for that provider first. Also renamed to `getListingByProvider`